packages/vite/src/node/plugins/esbuild.ts

```diff
@@ -493,15 +515,15 @@ function reloadOnTsconfigChange(changedFile: string) {
     server.moduleGraph.invalidateAll()
 
     // reset tsconfck so that recompile works with up2date configs
-    initTSConfck(server.config).finally(() => {
-      // server may not be available if vite config is updated at the same time
-      if (server) {
-        // force full reload
-        server.ws.send({
-          type: 'full-reload',
-          path: '*',
-        })
-      }
-    })
+    initTSConfck(server.config.root, true)
+
+    // server may not be available if vite config is updated at the same time
+    if (server) {
+      // force full reload
+      server.ws.send({
+        type: 'full-reload',
+        path: '*',
+      })
+    }
   }
```

Comment:
Since tsconfck is non-blocking now, we can reload the page right away, those that rely on tsconfig will wait for the promise before continuing.